src/modules/Elsa.Workflows.Runtime/Services/DefaultWorkflowDefinitionStorePopulator.cs

```diff
@@ -202,6 +166,67 @@ private async Task AddOrUpdateCoreAsync(MaterializedWorkflow materializedWorkflo
         }
         
         await _workflowDefinitionStore.SaveManyAsync(workflowDefinitionsToSave, cancellationToken);
+        return;
+
+        async Task UpdateIsLatest()
+        {
+            // Always try to update the IsLatest property based on the VersionNumber
+        
+            // Reset current latest definitions.
+            var filter = new WorkflowDefinitionFilter
+            {
+                DefinitionId = definitionId,
+                VersionOptions = VersionOptions.Latest
+            };
+            var latestWorkflowDefinitions = (await _workflowDefinitionStore.FindManyAsync(filter, cancellationToken)).ToList();
+
+            // If the latest definitions contains definitions with the same ID then we need to replace them with the latest workflow definitions.
+            SyncExistingCopies(latestWorkflowDefinitions, workflowDefinitionsToSave);
+
+            foreach (var latestWorkflowDefinition in latestWorkflowDefinitions)
+            {
+                if (latestWorkflowDefinition.Version > workflow.Identity.Version)
+                {
+                    _logger.LogWarning("A more recent version of the workflow has been found, overwriting the IsLatest property on the workflow.");
+                    existingWorkflowLatest = true;
+                    continue;
+                }
+
+                latestWorkflowDefinition.IsLatest = false;
+                workflowDefinitionsToSave.Add(latestWorkflowDefinition);
+            }
+        }
+
+        async Task UpdateIsPublished()
+        {
+            // If the workflow being added is configured to be the published version, then we need to reset the current published version.
+            if (workflow.Publication.IsPublished)
+            {
+                // Reset current published definitions.
+                var filter = new WorkflowDefinitionFilter
+                {
+                    DefinitionId = definitionId,
+                    VersionOptions = VersionOptions.Published
+                };
+                var publishedWorkflowDefinitions = (await _workflowDefinitionStore.FindManyAsync(filter, cancellationToken)).ToList();
+
+                // If the published workflow definitions contains definitions with the same ID as definitions in the latest workflow definitions, then we need to replace them with the latest workflow definitions.
+                SyncExistingCopies(publishedWorkflowDefinitions, workflowDefinitionsToSave);
+
+                foreach (var publishedWorkflowDefinition in publishedWorkflowDefinitions)
+                {
+                    if (publishedWorkflowDefinition.Version > workflow.Identity.Version)
+                    {
+                        _logger.LogWarning("A more recent version of the workflow has been found to be published, overwriting the IsPublished property on the workflow.");
```

Comment:
Log event messages should be fragments, not sentences. Avoid a trailing period/full stop.